### DISCUSSION TITLE
RK61

### DISCUSSION THREAD

**Discussion by skyisthela** [mat1jaczyyy/Keyboard-Inspector#28] [Help] [answered]
i have an rk61, i decided to try this software out on a whim, when i did a test the bottom left graph peaked at 50 something hz while the middle is consistently 1000hz and the right one is arbitrary, sometimes a 1000 sometimes 1300 or 1900 or whatever it's not consistent. i did another test only using NUMBERS and the left graph showed 990-999 hz consistently, while the middle and right showed random numbers.

**Comment by mat1jaczyyy** [comment]
This issues forum is meant for development of the software. For general usage questions the [Discord](https://discord.gg/kX4cJQH5Zn) is better suited.

That said, how the graphs are generated and what they mean in the input data is explained in technical detail [here](https://github.com/mat1jaczyyy/Keyboard-Inspector/blob/master/DOCUMENTATION.md#time-domain). The general takeaway though is that you probably need some fundamental understanding of signal analysis to be able to interpret these accurately. The [README](https://github.com/mat1jaczyyy/Keyboard-Inspector#getting-started) does talk about these in simpler terms, and I do think we should provide some more examples rather than theory talk in general.

It's hard to help you with your results directly if you don't share a .kbi file or at the very least a screenshot of the analysis.

**Comment by skyisthela** [comment]
> This issues forum is meant for development of the software. For general usage questions the [Discord](https://discord.gg/kX4cJQH5Zn) is better suited.
> 
> That said, how the graphs are generated and what they mean in the input data is explained in technical detail [here](https://github.com/mat1jaczyyy/Keyboard-Inspector/blob/master/DOCUMENTATION.md#time-domain). The general takeaway though is that you probably need some fundamental understanding of signal analysis to be able to interpret these accurately. The [README](https://github.com/mat1jaczyyy/Keyboard-Inspector#getting-started) does talk about these in simpler terms, and I do think we should provide some more examples rather than theory talk in general.
> 
> It's hard to help you with your results directly if you don't share a .kbi file or at the very least a screenshot of the analysis.

yeah mb man, also for the record i did read the readme, i still understood nothing lol
here's a screenshot of just randomly pressing letters on my board
![image](https://github.com/mat1jaczyyy/Keyboard-Inspector/assets/150245867/3e286cab-c9c2-43b7-a217-fe1862fa1b29)
here's a screenshot of using numbers 
![image](https://github.com/mat1jaczyyy/Keyboard-Inspector/assets/150245867/8780d75e-bbc2-4024-b2e7-28e83144ece2)

note that, it was still incredibly arbitrary results are different nearly all the time

**Comment by mat1jaczyyy** [comment]
If you get wildly different results every time, send a couple of different files. It's easier if you save the file from the File menu as I can then load it and play around with the analysis parameters.

Judging off this screenshot, looks like you have decent 1000Hz, but extremely severe chord splitting issues.

**Reply by skyisthela** [reply]
can you elaborate on what chord splitting is? heard of it but never really did any research

**Reply by skyisthela** [reply]
[KBI FILES.zip](https://github.com/mat1jaczyyy/Keyboard-Inspector/files/14050556/KBI.FILES.zip)
because i can't send kbi files i've zipped em

**Reply by mat1jaczyyy** [reply]
When you input more than one keypress at the same time, for example 4 keys for a 4-note chord in a rhythm game, you expect the keyboard to send all the keypresses in the same USB report.

Chord splitting happens when one of these keys is sent instantly, the second key on the next poll, the third key on the poll after that, etc. progressively delaying/splitting the chord more and more.

Your keyboard splits at ~4-5ms, which means the last keypress for a 4-note chord can be delayed up to 15ms. This causes accuracy issues in VSRG games

**Reply by mat1jaczyyy** [reply]
I'll have a look at the files in a bit, I'm away from the computer

**Reply by mat1jaczyyy** [reply, marked as answer]
@skyisthela you're getting wildly different results because you're not putting in a lot of key events, so the analysis is not too sure what to make of it. you're doing ~100, aim for 500-700 at least.

either way i think its enough data for me to conclude its good 1000hz but you have severe 5ms chordsplitting which makes this keyboard awful for acc based vsrg

**Reply by skyisthela** [reply]
thank you so much! is there any way to remove chord splitting? or is this an inherent issue on my board?

**Reply by mat1jaczyyy** [reply]
inherent issue with the firmware on the keyboard

**Reply by skyisthela** [reply]
alright thanks again. that kinda sucks because I just recently bought another keyboard from royal kludge. thanks again tho.

**Comment by mat1jaczyyy** [comment]
Also, is this Royal Kludge RK61? or

**Reply by skyisthela** [reply]
yes it's the Royal Kludge 61

**Reply by mat1jaczyyy** [reply]
Interesting, I've received files from others with the same keyboard but they had even worse results
